lib/AST/ASTImporter.cpp

```diff
@@ -1737,7 +1764,8 @@ Decl *ASTNodeImporter::VisitTypedefNameDecl(TypedefNameDecl *D, bool IsAlias) {
   ToTypedef->setAccess(D->getAccess());
   ToTypedef->setLexicalDeclContext(LexicalDC);
   Importer.Imported(D, ToTypedef);
-  LexicalDC->addDeclInternal(ToTypedef);
+  if (!isa<TranslationUnitDecl>(LexicalDC))
```

Comment:
I am not sure that this is the good solution. 
I would go with something like dynamic casting `D` to `TypeAliasDecl` and check for the result of `getDescribedAliasTemplate `.

Reply:
My first attempt was to use `if(!ToTypedef->getDescribedTemplate())`, but that simply did not pass.

Reply:
I realilzed now that we must check for the described template in the original decl (the one we import from). Updated.